src/util/sfdcUrl.ts

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { URL } from 'url';
+import { Env, Duration } from '@salesforce/kit';
+import { ensureNumber, ensureArray } from '@salesforce/ts-types';
+import { MyDomainResolver } from '../status/myDomainResolver';
+import { Logger } from '../logger';
+
+export class SfdcUrl extends URL {
+  /**
+   * Salesforce URLs
+   */
+  public static readonly SANDBOX = 'https://test.salesforce.com';
+  public static readonly PRODUCTION = 'https://login.salesforce.com';
+  public static readonly cache: Set<string> = new Set();
```

Comment:
This should be private.  I can't think of a reason why this would be useful outside of this class.

Reply:
I only needed access to the cache to clear it in the tests but, instead of doing that I just a different entry in the set